packages/realm-web/src/Fetcher.ts

```diff
@@ -64,14 +64,19 @@ function asyncIteratorFromResponseBody(body: unknown): AsyncIterable<Uint8Array>
   }
 }
 
+interface UserWithOptionalHeaders extends User {
+  /** extra headers defined by the user */
+  headers?: { [key: string]: string }; 
```

Comment:
I think `headers` should be added directly to the User class.

Reply:
I see pros and cons, and tensions between OO Typescript and Javascript that I alluded in my commit comment.  Referring to the 'O' in SOLID, my claim is: this feature is not intrinsic to existing App or User classes, but Fetcher only.  In support, ``LocationUrlContext``` is an invented trait/interface created in Fetcher module to overlay the App class; I created this special interface in the same spirit.  Changes in User module had only to do with passing constructor options to Fetcher.   On the con side, I think I agree that the feature should not be hidden & undocumented.

Reply:
One use case that I'm pondering is what if you want to store some header data when currentUser is set?  For example, you want to store a timestamp of when the user logged or establish a non-user specific session ID, in which you want included as header data in your requests.  I think being able to configure the headers for the user object before constructing the fetcher would be desirable in this case.

Reply:
Couldn't we consider your use case as property of the App instead of currentUser?  The way Fetcher objects are constructed and cloned through the App and User constructors gives it (the Fetcher) a reference to both.  If a property "headers" on User or "extraHeaders" on App exists at call time, Fetcher will forward those with the HTTP request.  The properties could be dynamic, a getter function.  There is no requirement of "before constructing the fetcher"

There is one case where the extra headers are static, this is in call to User.mongoClient (which is a sort of constructor).  The extra headers option becomes a property of the clone of User's fetcher & this is most accessible to the casual user.

I don't have a strong opinion on typing, but suggest adding an optional property (to be defined elsewhere) on Users means the same should be done on App.  Since Fetcher's only knowledge of its App dependency is through LocationUrlContext interface, the "extraHeaders" optional property should remain in the interface, but the UserContextWithOptionalHeaders is redundant.